### PR TITLE
Remove default values of DDS

### PR DESCRIPTION
**Pull request by tachikoma-li**
Remove unnecessary default values for the DDS class.
Also, remove the check for the number of offsets exceeds `UPPER_BOUND_OFFSETS` as I think this was for BLACK OPAL.

I think we can turn DDS into a dataclass and just remove the `__repr__` and `__str__`, which seem quite unnecessary here.

### CODE REVIEW

File: qctrlopencontrols/dynamic_decoupling_sequences/dynamic_decoupling_sequence.py [hunk 1]
```diff
@@ -49,18 +48,12 @@ class DynamicDecouplingSequence:
         The total time in seconds for the sequence :math:`\tau`.
     offsets : np.ndarray
         The times offsets :math:`\{t_j\}` in seconds for the center of pulses.
-    rabi_rotations : np.ndarray, optional
+    rabi_rotations : np.ndarray
         The rabi rotation :math:`\omega_j` at each time offset :math:`t_j`.
-        Defaults to ``None``.
-        If not provided, it will be set as :math:`\pi` at each time offset.
     azimuthal_angles : np.ndarray, optional
```
Comment (charmasaur) [inline qctrlopencontrols/dynamic_decoupling_sequences/dynamic_decoupling_sequence.py:53]:
```suggestion
    azimuthal_angles : np.ndarray
```

File: qctrlopencontrols/dynamic_decoupling_sequences/dynamic_decoupling_sequence.py [hunk 2]
```diff
@@ -49,18 +48,12 @@ class DynamicDecouplingSequence:
...
         The azimuthal angle :math:`\phi_j` at each time offset :math:`t_j`.
-        Defaults to ``None``.
-        If not provided, it will be set as 0 at each time offset.
     detuning_rotations : np.ndarray, optional
```
Comment (charmasaur) [inline qctrlopencontrols/dynamic_decoupling_sequences/dynamic_decoupling_sequence.py:55]:
```suggestion
    detuning_rotations : np.ndarray
```

File: qctrlopencontrols/dynamic_decoupling_sequences/dynamic_decoupling_sequence.py [hunk 3]
```diff
@@ -86,66 +79,51 @@ def __init__(
...
-            "Number of offsets is above the allowed number of maximum offsets. ",
-            {"offsets": offsets},
-            extras={
-                "number_of_offsets": len(offsets),
-                "allowed_maximum_offsets": UPPER_BOUND_OFFSETS,
-            },
-        )
-        check_arguments(
-            np.all(offsets >= 0.0) and np.all(offsets <= duration),
+            np.all((offsets >= 0) & (duration >= offsets)),
```
Comment (charmasaur) [inline qctrlopencontrols/dynamic_decoupling_sequences/dynamic_decoupling_sequence.py:96]:
```suggestion
            np.all(0 <= offsets <= duration),
```
Does that work?

**Review by charmasaur — CHANGES_REQUESTED** [review]

File: qctrlopencontrols/dynamic_decoupling_sequences/dynamic_decoupling_sequence.py (hunk 3 above)
Comment (tachikoma-li) [reply qctrlopencontrols/dynamic_decoupling_sequences/dynamic_decoupling_sequence.py:96]: nope, python would complain `The truth value of an array with more than one element is ambiguous. Use a.any() or a.all()`,
and I was quite surprised.

Comment (charmasaur) [reply qctrlopencontrols/dynamic_decoupling_sequences/dynamic_decoupling_sequence.py:96]: Ah, it's equivalent to `0 <= offsets and offsets <= duration` (https://docs.python.org/2.3/ref/comparisons.html), which would explain it. Oh well.

**Review by charmasaur — APPROVED** [review]

**Pull request merged**